pika/adapters/select_connection.py

```diff
@@ -937,12 +927,8 @@ def poll(self):
                         self._fd_events[PollEvents.ERROR],
                         self._get_max_wait())
                 else:
-                    # NOTE When called without any FDs, select fails on
-                    # Windows with error 10022, 'An invalid argument was
-                    # supplied'.
-                    time.sleep(self._get_max_wait())
-                    read, write, error = [], [], []
-
+                    # https://github.com/pika/pika/issues/1044#issuecomment-388514004
+                    raise RuntimeError('All _fd_events arrays are empty')
```

Comment:
By the way, my understanding is that `RuntimeError` is intended to be used as the base class of the Python runtime errors, not our errors encountered during runtime.

Reply:
I referred to [this documentation](https://docs.python.org/2/library/exceptions.html#exceptions.RuntimeError). It seemed like the best choice.